### ISSUE TITLE
The draw state validation layer reports that a timer query from 2 frames ago is unavailable or inflight.

### ISSUE THREAD

**Issue by karl-lunarg** [KhronosGroup/Vulkan-LoaderAndValidationLayers#367]
Moved from gitlab 99 (more discussion there)

The draw state validation layer reports that a timer query from 2 frames ago is unavailable, even though the command buffer the timer query was issued in has already been waited on and is being reused.
`"Cannot get query results on queryPool 129563503328 with index 4 which is unavailable."`
This error is still reported even when using VK_QUERY_RESULT_WAIT_BIT on vkGetQueryPoolResults.
When rendering from multiple threads I often also get:
`"Cannot get query results on queryPool 115919648784 with index 4 which is in flight."`

**Comment by lentinem** [comment]
This is quite old at this point. Simple testing seems to indicate this is fine but one of the reasons this was unresolved was failure to repro so it would be good to know if this issue is still showing up.

**Comment by mre4ce** [comment]
I verified that the issue still exists with latest master.

I've attached an executable that reproduces the problem to LunarXChange issue #526.

Thanks!

**Comment by lentinem** [comment]
Karl, is anyone on your end able to take a look at the repro?

**Comment by karl-lunarg** [comment]
Sure, I'll take a look.

**Comment by karl-lunarg** [comment]
I was able to repro this and I did some instrumentation and sent a lot of detail to Michael.  The basic problem is that a WaitForFences on fences that are in CB(s) unrelated to the queries is having an unwanted side effect on the query pool tracking in the validation layer.  The tracking is correct when the WaitForFences completes for the CB that had the WriteTimestamp in it that updated the query pool index  in question.  But a subsequent WaitForFences updated the query pool layer state with the state of a CB that wasn't using the query pool index updated by the WriteTimestamp in the previous CB.  The CB that wasn't using the query pool index (correctly) had this index (4) recorded as unavailable.  This "unavailable" state overwrote the "available" layer state from the previous CB

Returning to Michael who may have more knowledge of the design of this part of the layer.
